src/main/java/net/snowflake/ingest/streaming/internal/StreamingIngestStage.java

```diff
@@ -226,6 +235,18 @@ synchronized SnowflakeFileTransferMetadataWithAge refreshSnowflakeMetadata(boole
       }
 
       JsonNode responseNode = mapper.valueToTree(response);
+
+      // Update parameter map with the latest values from server
+      Map<String, Long> parameterMap =
+          mapper.convertValue(
+              responseNode.get("parameter_map"), new TypeReference<Map<String, Long>>() {});
```

Comment:
I think we want these for some boolean parameters as well, for example, BLOB_NO_HEADER, could we make it more generalize?

Reply:
And could you add a log line so we know what are those values if we need to look up customer's log during debugging issues

Reply:
I initially had the parameterMap be `Map<String, Object>` to be more generalizable but I ran into the issue where Jackson was creating all the `Long` values as `Integer`.  I didn't want there to be any confusion and type casting so I went with making all values `Long` and figured we could encode booleans with 0 or 1.  What do you think?

Reply:
I would still prefer to use Object instead of Long, I believe there is a way to tell Jackson to deserialize to Long instead of int, do you want to search around? If it doesn't work, please create a JIRA and I can try to see if there is anything i can do.